src/gs-category-page.c

```diff
@@ -266,6 +338,11 @@ gs_category_page_dispose (GObject *object)
 	g_clear_object (&self->subcategory);
 	g_clear_object (&self->plugin_loader);
 
+	if (self->category_apps != NULL) {
+		g_hash_table_unref (self->category_apps);
+		self->category_apps = NULL;
+	}
```

Comment:
Maybe just use `g_clear_pointer (&self->category_apps, (GDestroyNotify) g_hash_table_unref)` and remove the `if` condition ?

Reply:
:+1: